packages/mds-repository/filters.ts

```diff
@@ -0,0 +1,25 @@
+import { FindOperator, In, Equal, IsNull } from 'typeorm'
+
+export const PropertyValue = <TProperty extends string>(
+  property: TProperty,
+  value: unknown
+): Partial<{ [P in TProperty]: FindOperator<unknown> }> => {
+  return { [property]: Array.isArray(value) ? In(value) : Equal(value) } as Partial<
+    { [P in TProperty]: FindOperator<unknown> }
+  >
+}
+
+export const NullablePropertyValue = <TProperty extends string>(
+  property: TProperty,
+  value: unknown
+): Partial<{ [P in TProperty]: FindOperator<unknown> }> => {
+  return value === null
+    ? ({ [property]: IsNull() } as Partial<{ [P in TProperty]: FindOperator<unknown> }>)
+    : PropertyValue(property, value)
+}
+
+export const OptionalPropertyValue = <TProperty extends string>(
+  filter: (property: TProperty, value: unknown) => Partial<{ [P in TProperty]: FindOperator<unknown> }>,
```

Comment:
Is `filter` basically doing a type cast here? I don't really understand what's going on.

Reply:
No, it's the filter that's use if a property value is supplied. It would look like:
```ts
OptionalPropertyValue(PropertyValue, 'provider_id', provider_id)
```
Where `PropertyValue` would be called only when `provider_id` is `undefined`.